File: python/day2/part1.py
# A and X give rock, B and Y give paper, C and Z give scissors

points_for_rock = 1
points_for_paper = 2
points_for_scissors = 3

points_for_win = 6
points_for_draw = 3
points_for_loss = 0

player_score = 0


def translate_move_to_number(move):
    if move == "A" or move == "X":
        return points_for_rock
    elif move == "B" or move == "Y":
        return points_for_paper
    else:
        return points_for_scissors


def determine_winner(move_one, move_two):
    # if difference between the moves is 1, the first move wins, if difference is 2, second move wins
    if move_two - move_one == 1 or move_one - move_two == 2:
        return points_for_win
    elif move_two == move_one:
        return points_for_draw
    else:
        return points_for_loss


with open("text.txt") as file:
    for line in file:
        moves = line.split(" ")
        moves[1] = moves[1].strip()
        opponent_move = translate_move_to_number(moves[0])
        your_move = translate_move_to_number(moves[1])
        player_score += determine_winner(opponent_move, your_move) + your_move
    print(player_score)

File: java/day2/part1.java
import java.nio.file.Paths;
import java.util.ArrayList;
import java.util.Scanner;

public class PartOne {
    public static void main(String[] args) {
        int player_score = 0;

        String path = "text.txt";
        ArrayList<String> text = readFile(path);
        for (int i = 0, length = text.size(); i < length; i++) {
            String[] moves = text.get(i).split(" ");
            int opponentMove = translateMoveToNumber(moves[0]);
            int yourMove = translateMoveToNumber(moves[1]);
            player_score += determineWinner(opponentMove, yourMove) + yourMove;
        }
        System.out.println(player_score);
    }

    public static ArrayList<String> readFile(String path) {
        ArrayList<String> input = new ArrayList<>();
        
        try(Scanner scanner = new Scanner(Paths.get(path))) {
            while (scanner.hasNextLine()) {
                String line = scanner.nextLine();
                input.add(line);
            }
        } catch (Exception e) {
            System.out.println("Error: " + e.getMessage());
        }
        return input;
    }

    public static int translateMoveToNumber(String input) {
        int pointsForRock = 1;
        int pointsForPaper = 2;
        int pointsForScissors = 3;

        if (input.equals("A") || input.equals("X")) {
            return pointsForRock;
        } else if (input.equals("B") || input.equals("Y")) {
            return pointsForPaper;
        } else {
            return pointsForScissors;
        }
    }

    public static int determineWinner(int moveOne, int moveTwo) {
        int pointsForWin = 6;
        int pointsForDraw = 3;
        int pointsForLoss = 0;

        if (moveTwo - moveOne == 1 || moveOne - moveTwo == 2) {
            return pointsForWin;
        } else if (moveTwo == moveOne) {
            return pointsForDraw;
        } else {
            return pointsForLoss;
        }
    }
}
